### PR TITLE
Create order logic

### LINKED ISSUE #265: Add create order logic

### PR DESCRIPTION
**Pull request by D1nviz**
https://github.com/user-attachments/assets/5d1fe877-501a-4b81-9c9a-8caa651c2288

### CODE REVIEW

File: code/src/containers/order-summary/OrderSummary.scss [hunk 1]
```diff
@@ -1,48 +1,48 @@
...
-}
+@import "@design-system";
+
+.spa-cart-page {
+  &__order-summary {
+    width: 490px;
+
+    &--label {
+      margin-bottom: spacing(5);
+      font-size: $fs-h3;
```
Comment (mhevyk) [inline code/src/containers/order-summary/OrderSummary.scss]: Use `AppTypography`

File: code/src/containers/order-summary/OrderSummary.scss [hunk 2]
```diff
@@ -1,48 +1,48 @@
...
+      border-bottom: 1px solid
+        var(--spa-order-summary-border-color, $gray-light);
+      padding-bottom: spacing(3);
+    }
+
+    .spa-order-summary__details {
+      .spa-order-summary__text {
```
Comment (mhevyk) [inline code/src/containers/order-summary/OrderSummary.scss]: Fix nesting

File: code/src/containers/order-summary/OrderSummary.scss [hunk 3]
```diff
@@ -1,48 +1,48 @@
...
+        }
+      }
+
+      .spa-order-summary__underline-text {
+        text-decoration: underline;
+        margin: spacing(4) 0;
+      }
+
+      .spa-order-summary__total {
+        font-size: $fs-lg;
```
Comment (mhevyk) [inline code/src/containers/order-summary/OrderSummary.scss]: Same

File: code/src/containers/order-summary/OrderSummary.tsx [hunk 4]
```diff
@@ -1,78 +1,86 @@
...
+  isLoading
+}: OrderSummaryProps) => {
+  const formattedTotalPrice = formatPrice(totalPrice);
+
+  return (
+    <AppBox className="spa-cart-page__order-summary">
+      <AppTypography
+        className="spa-cart-page__order-summary--label"
+        variant="h1"
+        component="h1"
```
Comment (mhevyk) [inline code/src/containers/order-summary/OrderSummary.tsx:27]: Remove it

File: code/src/hooks/use-create-order/useCreateOrder.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    if (error) {
+      return;
+    }
+
+    openSnackbarWithTimeout({
+      messageTranslationKey: "orders.create.success",
+      variant: "success",
+      autohideDuration: 5000
+    });
+    navigate(routes.home.path);
```
Comment (mhevyk) [inline code/src/hooks/use-create-order/useCreateOrder.ts]: Add `{ replace: true } `option

File: mock-backend/src/controllers/orderConroller.js [hunk 6]
```diff
@@ -8,4 +8,16 @@ const getAdminOrders = (req, res) => {
   res.json(adminOrders);
 };
 
-module.exports = { getUserOrders, getAdminOrders };
+const createOrder = (req, res) => {
+  const { body } = req;
+  const newOrder = {
+    id: Math.floor(Math.random() * 1000),
```
Comment (mhevyk) [inline mock-backend/src/controllers/orderConroller.js]: Use `crypto.randomUUID()` instead or better `Date.now()`

File: mock-backend/src/controllers/orderConroller.js [hunk 7]
```diff
@@ -8,4 +8,16 @@ const getAdminOrders = (req, res) => {
...
+    ...body,
+  };
+
+  setTimeout(() => {
+    res.json(newOrder.id);
+  }, 1000);
```
Comment (mhevyk) [inline mock-backend/src/controllers/orderConroller.js]: Use `wait` util

File: code/src/containers/order-summary/OrderSummary.scss (hunk 1 above)
Comment (D1nviz) [reply code/src/containers/order-summary/OrderSummary.scss]: used

File: code/src/containers/order-summary/OrderSummary.scss (hunk 2 above)
Comment (D1nviz) [reply code/src/containers/order-summary/OrderSummary.scss]: fixed

File: code/src/containers/order-summary/OrderSummary.scss (hunk 3 above)
Comment (D1nviz) [reply code/src/containers/order-summary/OrderSummary.scss]: also used

File: code/src/containers/order-summary/OrderSummary.tsx (hunk 4 above)
Comment (D1nviz) [reply code/src/containers/order-summary/OrderSummary.tsx:27]: refactored

File: code/src/hooks/use-create-order/useCreateOrder.ts (hunk 5 above)
Comment (D1nviz) [reply code/src/hooks/use-create-order/useCreateOrder.ts]: added

File: mock-backend/src/controllers/orderConroller.js (hunk 6 above)
Comment (D1nviz) [reply mock-backend/src/controllers/orderConroller.js]: used `Date.now()`

File: mock-backend/src/controllers/orderConroller.js (hunk 7 above)
Comment (D1nviz) [reply mock-backend/src/controllers/orderConroller.js]: used

**Review by OliaHrytsak — APPROVED** [review]
Good job!

**Review by mhevyk — APPROVED** [review]
Good job!

**Comment by D1nviz** [comment]
Thank you for your comments. I appreciate it! 😊

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**